engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/TableLocationKeyFinder.java

```diff
@@ -14,6 +14,17 @@
 @FunctionalInterface
 public interface TableLocationKeyFinder<TLK extends TableLocationKey> {
 
+    /**
+     * May create a new {@link TableLocationKeyFinder} with additional safety checks.
+     *
+     * @param finder the finder
+     * @return the finder, potentially with additionaly safety checks
+     * @param <TLK> the table location key type
+     */
+    static <TLK extends TableLocationKey> TableLocationKeyFinder<TLK> safetyCheck(TableLocationKeyFinder<TLK> finder) {
+        return new TableLocationKeySafetyImpl<>(finder);
+    }
```

Comment:
Some JavaDoc cleanup:
```suggestion
    /**
     * May create a new TableLocationKeyFinder with additional safety checks.
     *
     * @param finder The finder
     * @return The finder, potentially with additional safety checks added
     * @param <TLK> The {@link TableLocationKey} type
     */
    static <TLK extends TableLocationKey> TableLocationKeyFinder<TLK> safetyCheck(
            @NotNull TableLocationKeyFinder<TLK> finder) {
        return new TableLocationKeySafetyImpl<>(finder);
    }
```